app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PostsController < ApplicationController
+
+  def index
+    @post = Post.new
+    @comment = Comment.new
+    @posts = current_user.all_friend_posts
+  end
+
+  def show
+    @post = Post.with_comments_and_likes.find(params[:id])
+    @comment = Comment.new
+    @post = Post.new
```

Comment:
Which of the two between lines 10 and 12 decide on what `@post` is gonna be? The way it is written now it will appear that line 10 is unnecessary and we can safely delete it.

Reply:
It was a mistake on my part, I have deleted line 12 as it is unnecessary in this case.